File: joo/0511/pro_0511_1.py
def solution(cap, n, deliveries, pickups):
    answer = 0
    deliveries = deliveries[::-1]
    pickups = pickups[::-1]
    deli = 0
    pick = 0
    for i in range(n):
        deli += deliveries[i]
        pick += pickups[i]

        while deli > 0 or pick > 0:
            deli -= cap
            pick -= cap
            answer += (n - i) * 2

    return answer

File: joo/0511/pro_0511_1.java
// 프로그래머스 2023 kakao blind recruitment - 택배 배달과 수거하기
// https://school.programmers.co.kr/learn/courses/30/lessons/150369?language=java

package al_0511;

public class pro_0511_1 {
	class Solution {
	    public long solution(int cap, int n, int[] deliveries, int[] pickups) {
	        long answer = 0;
	        int deli = 0;
	        int pick = 0;
	        for (int i = n-1; i >= 0; i--) {
	            deli += deliveries[i];
	            pick += pickups[i];
	            
	            while (deli > 0 || pick > 0) {
	                deli -= cap;
	                pick -= cap;
	                answer += ((i+1) * 2);
	            }
	        }
	        return answer;
	    }
	}

}
